vscode-extension/src/providers/textDocumentContentProvider.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import * as path from 'path';
+import * as fs from 'fs';
+import {TextDocumentContentProvider, Event, Uri, ProviderResult} from 'vscode';
+import {RelayExtensionContext} from '../context';
+
+const EMPTY_JSON_SCHEMA = '{}';
+const RELAY_CONFIG_SCHEMA_PATH = 'relay-config-schema';
+const PACKAGE_JSON_RELAY_CONFIG_SCHEMA_PATH =
+  'package-json-relay-config-schema';
+
+export class RelayTextDocumentContentProvider
+  implements TextDocumentContentProvider
+{
+  private readonly jsonSchemaPath: string;
+
+  constructor(context: RelayExtensionContext) {
+    this.jsonSchemaPath = path.join(
+      context.relayBinaryExecutionOptions.binaryPath,
+      '../../config-schema.json',
+    );
+  }
+
+  static readonly scheme: string = 'relay';
+
+  onDidChange?: Event<Uri> | undefined;
+
+  provideTextDocumentContent(uri: Uri): ProviderResult<string> {
+    if (uri.authority === PACKAGE_JSON_RELAY_CONFIG_SCHEMA_PATH) {
+      return `{"properties": { "relay": { "$ref": "${RelayTextDocumentContentProvider.scheme}://${RELAY_CONFIG_SCHEMA_PATH}", "description": "Relay.js configuration" }}}`;
+    }
+
+    if (uri.authority === RELAY_CONFIG_SCHEMA_PATH) {
+      try {
+        return fs.readFileSync(this.jsonSchemaPath, {encoding: 'utf-8'});
```

Comment:
Oder versions of the compiler will not have this file and might error here. Should we check the compiler's version before expecting this to exist?

Reply:
I've added a check whether or not the package version of `relay-compiler` is >=14.2, since I'm assuming that will be the version this feature might be released in.